macos/library/FLEFileChooserPlugin.m

```diff
@@ -30,11 +30,20 @@
 static NSString *const kPlatformMethodNameKey = @"method";
 static NSString *const kPlatformMethodArgsKey = @"args";
 
-// Method argument keys.
+// Appearance configuration keys for NSSavePanel.
+static NSString *const kPanelTitleKey = @"title";
+static NSString *const kPanelMessageKey = @"message";
+static NSString *const kOKButtonLabelKey = @"prompt";
+
+// Method argument keys for NSSavePanel.
+static NSString *const kInitialDirectoryKey = @"initialDirectory";
 static NSString *const kAllowedFileTypesKey = @"allowedFileTypes";
+static NSString *const kInitialFileName = @"nameFieldStringValue";
+
+// Method argument keys for NSOpenPanel.
 static NSString *const kAllowsMultipleSelectionKey = @"allowsMultipleSelection";
 static NSString *const kCanChooseDirectoriesKey = @"canChooseDirectories";
-static NSString *const kInitialDirectoryKey = @"initialDirectory";
+static NSString *const kCanChooseFilesKey = @"canChooseFiles";
```

Comment:
Given that for our Linux implementation it would be a hack to allow both files and directories, let's just document the Dart side to make it clear that canChooseDirectories enables directories but disables files (and change the the macOS plugin to match that behavior).

We can rename it later, once the Dart code is in this repo.

If at some point someone has a specific use case for this, we can revisit doing a hack on Linux (adding another button to a file chooser that allows folders seems to be an option).

Reply:
:+1:

Reply:
Done. I will create an issue to update the dart side with the suggested documentation, and to remove references to parameter keys that are not defined in the macOS plugin (title, message, canChooseFiles).